src/Maestro/Microsoft.DotNet.Maestro.Tasks.Tests/GetManifestAsAssetTests.cs

```diff
@@ -0,0 +1,204 @@
+        {
+            AssetData expectedPackageAssetNoLocationOrManifest =
+                new AssetData(true)
+                {
+                    Locations = ImmutableList.Create(
+                        new AssetLocationData(LocationType.Container)
+                        { Location = "" }),
+                    Name = "assets/manifests/thisIsARepo/1.2.3-beta.1235.6/",
+                    Version = "1.2.3-beta.1235.6"
+                };
+
+            AssetData parsedAssets = pushMetadata.GetManifestAsAsset(ImmutableList.Create(PackageAsset1), "", "");
+            parsedAssets.Should().BeEquivalentTo(expectedPackageAssetNoLocationOrManifest);
+        }
+
+        [Test]
+        public void GivenAssetVersionIsSet_VersionDoesntChange()
+        {
+            AssetData packageAsset1DifferentVersion =
+                new AssetData(true)
+                {
+                    Locations = ImmutableList.Create(
+                        new AssetLocationData(LocationType.Container)
+                        { Location = locationString }),
+                    Name = "PackageAsset1Name",
+                    Version = "123456"
+                };
+
+            // Version is set on the first pass through the function, then persisted in the class instance so changing it should have no effect on the generated AssetData
+            pushMetadata.GetManifestAsAsset(ImmutableList.Create(PackageAsset1), "thisIsALocation", "FirstManifest");
+            AssetData secondPassAssets = pushMetadata.GetManifestAsAsset(ImmutableList.Create(packageAsset1DifferentVersion), locationString, newManifestName);
+            secondPassAssets.Should().BeEquivalentTo(ExpectedPackageAsset1);
+        }
+
+        [Test]
+        public void GivenAssetVersionNotSetAndNonShippingAssetWithoutVersion()
+        {
+            AssetData packageAssetNoVersion =
+                new AssetData(true)
+                {
+                    Locations = ImmutableList.Create(
+                        new AssetLocationData(LocationType.Container)
+                        { Location = locationString }),
+                    Name = "PackageAsset1Name"
+                };
+
+            AssetData expectedPackageAssetNoVersion =
+                new AssetData(true)
+                {
+                    Locations = ImmutableList.Create(
+                        new AssetLocationData(LocationType.Container)
+                        { Location = locationString }),
+                    Name = $"assets/manifests/thisIsARepo//{newManifestName}"
+                };
+
+            AssetData parsedAssets = pushMetadata.GetManifestAsAsset(ImmutableList.Create(packageAssetNoVersion), locationString, newManifestName);
+            parsedAssets.Should().BeEquivalentTo(expectedPackageAssetNoVersion);
+        }
+
+        [Ignore("Returned AssetData is always labeled as NonShipping regardless of the input, is this intentional?")]
```

Comment:
Yes, this is correct. The merged manifest must always be unique build to build, and such is a non-shipping asset. You can remove this test.

Reply:
@meganaquinn Actually I think we should alter the code  in GetManifestAsAsset to error on this case, where there is no formally provided AssetVersion (this is usually passed in), and no non-shipping assets. In this case, the merged manifest will always be written to the wrong location.

Reply:
Could you file an issue for that since it isn't the current behavior?